File: old/python/A+B-linkedlist/naive.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


# My solution: naive.
class Solution:
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        s = self.getSum(l1) + self.getSum(l2)
        return self.makeList(s)

    def getSum(self, li):
        node = li
        mySum, exp = 0, 1
        while (node):
            mySum += node.val * exp
            node = node.next
            exp *= 10
        return mySum

    def makeList(self, total):
        nums = []
        if total == 0:
            return ListNode(0)
        while total:
            nums.append(total % 10)
            total //= 10
        nodes = [ListNode(num) for num in nums]
        for i in range(len(nums) - 1):
            nodes[i].next = nodes[i + 1]
        return nodes[0]

File: old/C++/Linkedlist/2-A+B-Polynomial/naive.cpp
#include <iostream>
using namespace std;

// Definition for singly-linked list.
struct ListNode
{
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution
{
  public:
    ListNode *addTwoNumbers(ListNode *l1, ListNode *l2)
    {
        int carry = 0, sum = 0;
        ListNode *head = new ListNode(0), *last = head, *current = NULL;
        while (l1 && l2)
        {
            sum = l1->val + l2->val + carry;
            carry = sum / 10;
            sum %= 10;

            current = new ListNode(sum);
            last->next = current;
            last = current;

            l1 = l1->next;
            l2 = l2->next;
        }
        ListNode *temp = NULL;
        if (l1)
            temp = l1;
        else
            temp = l2;
        while (temp)
        {
            sum = carry + temp->val;
            carry = sum / 10;
            sum %= 10;
            current = new ListNode(sum);
            last->next = current;
            last = current;
            temp = temp->next;
        }
        if (carry)
            last->next = new ListNode(carry);
        return head->next;
    }
};